### PR TITLE
[FIX] account: Fix bug if internal payment has 0 amount (while confirming payment)

### PR DESCRIPTION
**Pull request by RachidAlassir**
Description of the issue/feature this PR addresses:

Create a payment of type internal and with amount=0
Try to confirm payment and this error will be displayed:
"if not (all_accounts[0].reconcile or all_accounts[0].internal_type == 'liquidity'):
IndexError: list index out of range"

Screenshot:
image

Current behavior before PR:
Error appears after confirming an internal payment with amount=0
Desired behavior after PR is merged:
Check lines to reconcile before calling function `_check_reconcile_validity`.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by RachidAlassir** [comment]
@mart-e FYI
Also I tried to write as few as possible of code and I'm open for variable renaming like `lines_to_reconcile`
I also signed my CLA in this MR [61362](https://github.com/odoo/odoo/pull/61362) it's on its way to be merged.
Thanks

File: addons/account/models/account_move.py [hunk 1]
```diff
@@ -3847,7 +3847,8 @@ def reconcile(self, writeoff_acc_id=False, writeoff_journal_id=False):
         )
 
         reconciled_lines = self.filtered(lambda aml: float_is_zero(aml.balance, precision_rounding=aml.move_id.company_id.currency_id.rounding) and aml.reconciled)
-        (self - reconciled_lines)._check_reconcile_validity()
+        if self - reconciled_lines:
+            (self - reconciled_lines)._check_reconcile_validity()
```
Comment (smetl) [inline addons/account/models/account_move.py:3851]: I prefer adding ```
if not self:
    return
``` at the beginning of the `_check_reconcile_validity` method

**Review by smetl — CHANGES_REQUESTED** [review]

**Comment by RachidAlassir** [comment]
@smetl , Changes were made at the same commit (Commit updated), I also tested it on the runbot.
Thanks.

**Review by RachidAlassir — COMMENTED** [review]
@smetl , Changes were made at the same commit (Commit updated), I also tested it on the runbot.
Thanks.

**Comment by smetl** [comment]
@robodoo r+
@RachidAlassir Thanks!

**Comment by RachidAlassir** [comment]
rebase-ff

**Comment by RachidAlassir** [comment]
@robodoo rebase-ff

**Comment by RachidAlassir** [comment]
@smetl Please add the merge method, I think It's rebase-ff

**Comment by smetl** [comment]
@robodoo rebase-ff
@RachidAlassir sorry

**Pull request closed without merging**